fury/shaders/tests/test_base.py

```diff
@@ -97,26 +158,37 @@ def generate_cube_with_effect():
     return cube
 
 
-def test_shader_to_actor(interactive=False):
-    cube = generate_cube_with_effect()
+def generate_points():
+    centers = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
 
-    scene = window.Scene()
-    scene.add(cube)
-    if interactive:
-        scene.add(actor.axes())
-        window.show(scene)
+    colors = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]]) * 255
 
-    arr = window.snapshot(scene)
-    report = window.analyze_snapshot(arr)
-    npt.assert_equal(report.objects, 1)
+    vtk_vertices = Points()
+    # Create the topology of the point (a vertex)
+    vtk_faces = CellArray()
+    # Add points
+    for i in range(len(centers)):
+        p = centers[i]
+        id = vtk_vertices.InsertNextPoint(p)
+        vtk_faces.InsertNextCell(1)
+        vtk_faces.InsertCellPoint(id)
+    # Create a polydata object
+    polydata = PolyData()
+    # Set the vertices and faces we created as the geometry and topology of the
+    # polydata
+    polydata.SetPoints(vtk_vertices)
+    polydata.SetVerts(vtk_faces)
 
-    # test errors
-    npt.assert_raises(ValueError, shader_to_actor, cube, "error",
-                      vertex_impl)
-    npt.assert_raises(ValueError, shader_to_actor, cube, "vertex",
-                      vertex_impl, block="error")
-    npt.assert_raises(ValueError, replace_shader_in_actor, cube, "error",
-                      vertex_impl)
+    set_polydata_colors(polydata, colors)
+
+    mapper = PolyDataMapper()
+    mapper.SetInputData(polydata)
+    mapper.SetVBOShiftScaleMethod(False)
+
+    point_actor = Actor()
+    point_actor.SetMapper(mapper)
+
```

Comment:
Why restart from scratch and not use a point actor?   

To reduce the maintenance, you should try to reduce as much as possible code duplication